makeflow/src/makeflow_log.c

```diff
@@ -200,6 +224,42 @@ void makeflow_log_recover(struct dag *d, const char *filename, int verbose_mode,
 				}
 				continue;
 			}
+			if(sscanf(line, "# CACHE %" SCNu64 " %s", &previous_completion_time, cache_dir) == 2) {
+				/* if the user specifies a cache dir using --cache dir, ignore the info from the log file */
+				if(!d->cache_dir) {
+					d->cache_dir = xxstrdup(cache_dir);
+					if(!d->cache_dir) {
+						LDEBUG("xxstrdup(%s) failed!\n", cache_dir);
+						return;
+					}
+				}
+				continue;
+			}
+			if(sscanf(line, "# MOUNT %" SCNu64 " %s %s %s %d", &previous_completion_time, file, source, cache_name, &type) == 5) {
+				char *p = NULL;
+				f = dag_file_lookup_or_create(d, file);
+
+				/* if the user specifies the same target in the mountfile, ignore the info from the log file */
+				if(f->source) continue;
+
+				p = xxstrdup(source);
+				if(!p) {
```

Comment:
I am not sure that this checking is necessary as xxstrdup is used. If strdup fails then xxstrdup emits a fatal warning. @btovar am I misinterpreting xxstrdup?